Dz2

aagaryov74 opened the pull request:
It seems done

Commit:
change context creation with XML to annotations

Commit:
add properties and localization

Commit:
add properties and localization into test

stvort commented on src/main/java/ru/otus/agaryov/dz2/Main.java:
```diff
@@ -1,27 +1,52 @@
+                err.println("IO error:");
                 e.printStackTrace();
             }
         } else {
```
Проверку лучше сделать методом executor. Причем лучше ее вызывать внутри doExam. try/catch тоже лучше убрать внутрь. Вместо можно стектрейса вывести сообщение об ошибке. 
"Sorry but you cannot continue due errors with config file" лучше тоже локализовать

stvort commented on src/main/java/ru/otus/agaryov/dz2/csvfilereader/CsvFileReader.java:
```diff
@@ -7,4 +7,6 @@
     // How many correct strings are in config file?
     Integer getReadedStrsCount();
+    // change file if we need to change locale in runtime;
+    LinkedHashMap<String,String> readAnotherFile(String fileName);
```
Лучше сделать возвращаемым типом просто Map<String,String>. Тогда внутри метода будет легче менять реализацию.

stvort commented on src/main/java/ru/otus/agaryov/dz2/csvfilereader/ImplCsvFileReader.java:
```diff
@@ -36,6 +42,12 @@ public ImplCsvFileReader(String fileName) {
     }
 
+    @Override
+    public LinkedHashMap<String,String> readAnotherFile(String fileName) {
```
Достаточно просто сделать сеттер для csvFile

aagaryov74 replied:
Здравствуйте - тут не понял - задавать csvFile  мало - надо еще мапу перечитать из него.
Или комментарий касался названия метода?

stvort replied:
Ну я про то, что метод для чтения Map уже есть. Если сделать сеттер, то потом достаточно вызвать readCsvIntoMap

stvort commented on src/main/java/ru/otus/agaryov/dz2/exam/ExamExecutor.java:
```diff
@@ -1,37 +1,66 @@
 
-        out.println("Введите Ваше Имя и фамилию:");
+        Locale currentLocale = Locale.getDefault();
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
```
Работу с консолью (ввод/вывод) лучше завернуть в свой сервис. Запрос имени пользователя тоже можно сделать сервисом

stvort commented on src/main/java/ru/otus/agaryov/dz2/exam/ExamExecutor.java:
```diff
@@ -1,37 +1,66 @@
+        out.println(ms.getMessage("enterFio",null,currentLocale));
         String studentFIO = br.readLine();
-        out.printf("Здравствуйте, %s!\n Вам предстоит ответить на %d вопр.\n Начнем!\n\n",
+        if (asciiChecker.isASCII(studentFIO)) {
```
Локаль лучше указать в настройках и работать с ней.

stvort commented on src/main/resources/application_ru_RU.properties:
```diff
@@ -0,0 +1,7 @@
+#! CSV FILE NAME
```
Файлы настроек application.properties и файлы для хранения локализованных сообщений это должны быть разные файлы

Commit:
Проверку лучше сделать методом executor. Причем лучше ее вызывать внутри doExam. try/catch тоже лучше убрать внутрь. Вместо можно стектрейса вывести сообщение об ошибке.
"Sorry but you cannot continue due errors with config file" лучше тоже локализовать

Commit:
Лучше сделать возвращаемым типом просто Map<String,String>. Тогда внутри метода будет легче менять реализацию.

Commit:
stvort 11 hours ago
Достаточно просто сделать сеттер для csvFile

Commit:
stvort 12 hours ago
Работу с консолью (ввод/вывод) лучше завернуть в свой сервис. Запрос имени пользователя тоже можно сделать сервисом

Commit:
stvort 13 hours ago
Файлы настроек application.properties и файлы для хранения локализованных сообщений это должны быть разные файлы

stvort commented on src/main/java/ru/otus/agaryov/dz2/Main.java:
```diff
@@ -1,33 +1,50 @@
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(Main.class);
         CsvFileReader csvFile = context.getBean(CsvFileReader.class);
         ResultChecker checker = context.getBean(ImplResultChecker.class);
         ExamExecutor executor = context.getBean(ExamExecutor.class);
```
Не нужно вытаскивать другие бины, кроме executor

stvort commented on src/main/java/ru/otus/agaryov/dz2/service/IOService.java:
```diff
@@ -0,0 +1,70 @@
+        locale = Locale.getDefault();
+    }
+
+    public void outCons(String propParam) {
```
Не нужно сокращать названия методов, полей и переменных. В java являются совершенно нормальными названия из трех и более слов. В данном случае остается только догадываться, что Cons это консоль. И если это так, то мы теряем универсальность названия и не сможем выделить интерфейс (а по хорошему так и нужно сделать).

stvort commented on src/main/java/ru/otus/agaryov/dz2/service/IOService.java:
```diff
@@ -0,0 +1,70 @@
+
+    public Boolean checkAndSwitchLocale(String inpStr) {
+        boolean isLocaleChanged = false;
+        if (asciiCheckerService.isASCII(inpStr)) {
```
И все же лучше задать локаль в настройках или давать выбор пользователю. Данным кодом вы зафиксировали в своем приложении 2 языка. Чтобы добавить новый язык придется дописывать класс.

aagaryov74 replied:
настройки локали и так берутся из настроек системы. Я спрошу у пользователя хочет ли он менять локаль от системной и на какую в случае если он при аглийской локали вдруг представился не в ASCII

stvort commented on src/main/resources/application_ru_RU.properties:
```diff
@@ -0,0 +1,3 @@
+#! CSV FILE NAME
```
application.properties лучше сделать единственным файлом (не bundle). А вместо имени файла можно указать шаблон, по которому на лету формировать реальное имя с помощью текущей локали

Commit:
stvort 36 minutes ago
Не нужно вытаскивать другие бины, кроме executor

Commit:
stvort an hour ago
Не нужно сокращать названия методов, полей и переменных. В java являются совершенно нормальными названия из трех и более слов. В данном случае остается только догадываться, что Cons это консоль. И если это так, то мы теряем универсальность названия и не сможем выделить интерфейс (а по хорошему так и нужно сделать).

Commit:
stvort 3 hours ago
application.properties лучше сделать единственным файлом (не bundle). А вместо имени файла можно указать шаблон, по которому на лету формировать реальное имя с помощью текущей локали

Commit:
stvort 4 hours ago
И все же лучше задать локаль в настройках или давать выбор пользователю. Данным кодом вы зафиксировали в своем приложении 2 языка. Чтобы добавить новый язык придется дописывать класс.

  @aagaryov74
aagaryov74 3 hours ago  •
 Author Owner
настройки локали и так берутся из настроек системы. Я спрошу у пользователя хочет ли он менять локаль от системной и на какую в случае если он при аглийской локали вдруг представился не в ASCII

 @aagaryov74
